File: 2016/gara3/yoda/att/yoda.c
#include <stdio.h>
#include <string.h>
#include <assert.h>

void yodizza(char S[], char Y[]) {
    // Per calcolare la lunghezza si puo' fare cosi'
    int lunghezza = strlen(S);

    // Mettete qui il codice della soluzione
    int i;
    for (i=0; i<lunghezza; i++) {
         Y[i] = S[i];
    }
}


#define MAXS 100000

char S[MAXS + 1], Y[MAXS + 10];

int main() {
    FILE *fr, *fw;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(NULL != fgets(S, MAXS + 1, fr));

    yodizza(S, Y);

    fprintf(fw, "%s\n", Y);
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara3/yoda/att/yoda.pas
procedure yodizza(var S: string; var Y: array of char);
var lunghezza, i: longint;
begin
    (* Per calcolare la lunghezza si puo' fare cosi' *)
    lunghezza := length(S);

    (* Mettete qui il codice della soluzione *)
    for i:=0 to lunghezza - 1 do
        Y[i] := S[i + 1];  (* ATTENZIONE: le due stringhe sono indicizzate diversamente:  *)
                           (*             S parte da 1 mentre Y parte da 0. Viva Pascal!  *)
end;


const
    MAXS = 100000;
var
    N     : longint;
    S     : string;
    Y     : array[0..MAXS+10] of char;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, S);

    yodizza(S, Y);

    writeln(fw, Y);
    close(fr);
    close(fw);
end.
